"field" and "exclude" arguments of model_to_dict() do not accept sets

# Bug report

## What's wrong

Our test suite contains code that simplifies to this fragment:
```py
from typing import Mapping

from django.db.models.base import Model
from django.forms import model_to_dict


def check(instance: Model, data: Mapping[str, object]) -> None:
    assert data == model_to_dict(instance, fields=data.keys())
```

When checking that with mypy, it reports:
```
testcase.py:8: error: Argument "fields" to "model_to_dict" has incompatible type "AbstractSet[str]";
expected "Union[List[Union[Callable[..., Any], str]], Sequence[str], Literal['__all__'], None]"
[arg-type]
        assert data == model_to_dict(instance, fields=data.keys())
```

## How is that should be

The implementation of `model_to_dict()` only needs `__bool__()` and `__contains__()` to be provided by the `fields` and `exclude` arguments, so passing a keys set should not be flagged as an error.

I think a solution could be to replace `Sequence` in the stubs annotation with `Collection`.

## System information

- OS: Ubuntu Linux 18.04
- `python` version: 3.6.9
- `django` version: 2.2.1
- `mypy` version: 0.812
- `django-stubs` version: 1.8.0

--- a/django-stubs/forms/models.pyi
+++ b/django-stubs/forms/models.pyi
@@ -37,7 +37,7 @@ from django.db.models import ForeignKey
 
 ALL_FIELDS: str
 
-_Fields = Union[List[Union[Callable, str]], Collection[str], Literal["__all__"]]
+_Fields = Union[Collection[str], Literal["__all__"]]
 _Labels = Dict[str, str]
 _ErrorMessages = Dict[str, Dict[str, str]]
 

--- a/django-stubs/forms/models.pyi
+++ b/django-stubs/forms/models.pyi
@@ -2,6 +2,7 @@ from datetime import datetime
 from typing import (
     Any,
     Callable,
+    Collection,
     Dict,
     Iterator,
     List,
@@ -36,7 +37,7 @@ from django.db.models import ForeignKey
 
 ALL_FIELDS: str
 
-_Fields = Union[List[Union[Callable, str]], Sequence[str], Literal["__all__"]]
+_Fields = Union[List[Union[Callable, str]], Collection[str], Literal["__all__"]]
 _Labels = Dict[str, str]
 _ErrorMessages = Dict[str, Dict[str, str]]
 

--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ with open("README.md") as f:
     readme = f.read()
 
 dependencies = [
-    "mypy>=0.790",
+    "mypy>=0.790,<0.900",
     "typing-extensions",
     "django",
     "django-stubs-ext",

--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ with open("README.md") as f:
     readme = f.read()
 
 dependencies = [
-    "mypy>=0.790,<0.900",
+    "mypy>=0.900",
     "typing-extensions",
     "django",
     "django-stubs-ext",

